src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,225 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(final Car car) {
+        String query = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query,
+                         Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create car: " + car + ".", e);
+        }
+        if (car.getId() != null) {
+            addDrivers(car.getId(), car.getDrivers());
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(final Long carId) {
+        Optional<Car> carResult = Optional.empty();
+        String query = "SELECT c.id AS car_id, model AS car_model, manufacturer_id AS m_id, "
+                + "name AS m_name, country AS m_country FROM cars c "
+                + "JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id "
+                + "WHERE c.id = ? AND c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, carId);
+            statement.executeQuery();
+            ResultSet resultSet = statement.getResultSet();
+            if (resultSet.next()) {
+                Car car = new Car();
+                car.setId(carId);
+                car.setModel(resultSet.getString(2));
+                car.setManufacturer(generateManufacturer(resultSet));
+                carResult = Optional.of(car);
```

Comment:
Why aren't you using your generateCar method here?

Reply:
Done